### PR TITLE
1.1.5 worked example organization classifications

### LINKED ISSUE #711: Parties: Organization classifications (e.g. women-owned businesses, legal entity type)
## Motivation

Some procurement policies create obligations to track and/or promote the participation of women-owned businesses in contracting processes.

Many other classifications exist; women-owned businesses is just used as a common example.

## Discussion

Small or Medium Enterprise (SME) is another classification that is frequently the subject of procurement policies. There is a community extension to distinguish SMEs: https://github.com/open-contracting/ocds_partyDetails_scale_extension

For 'women-owned' and other ownership-related classifications, company registers rarely if ever track this information as structured data. All known cases of structured data are from procurement systems.

The US DATA Act Information Model Schema (DAIMS) tags awards with different facts ("Procurement Award Specific Characteristics"), like whether it's "Minority Owned Business" or "Woman Owned Business" but also whether it's "Airport Authority" or "Clinger-Cohen Act Planning Compliance". See [this diagram](https://fedspendingtransparency.github.io/assets/docs/DAIMS_IDD_Diagram_File_D1_v1.1.pdf) from [this page](https://fedspendingtransparency.github.io/DAIMS-v1.1/).

If the system were storing demographic facts about the organization's control in a separate field, we could create an extension with narrowly defined fields. However, given that known systems store various facts in a single field, we should create an extension with broadly defined fields, closer to "tags." The extension could have an open codelist to allow a standard way of expressing e.g. "woman-owned business".

## Proposal

Given that the classifications of organizations may be about ownership, incorporation, sector, location, number of employees, etc., and given that these classifications intersect ("Small Agricultural Cooperative"), I propose reusing the Classification object. The object is currently only used for Item classifications. Its `scheme` field uses an itemClassificationScheme codelist, but otherwise the object is generic. As I have no reason to believe that organizations have a 'primary' classification, I propose naming the field `classifications` instead of following the pattern of a primary `classification` and then `additionalClassifications`. ~~I also recommend deprecating the Party Scale extension as its use case will be duplicated by this extension.~~

Example:

```json
{
  "name": "Acme Inc.",
  "classifications": [{
    "id": "sme",
    "description": "petite et moyenne entreprise"
  }, {
    "id": "women-owned",
    "description": "entreprise appartenant à des femmes"
  }]
}
```

In terms of existing schemes:

* As mentioned, DAIMS has a codelist, but it mixes organization and non-organization classifications.
* Open Civic Data has an [organization classification codelist](https://github.com/opencivicdata/python-opencivicdata/blob/master/opencivicdata/common.py#L63).
* Jurisdictions have a limited set of registered company types (SA, PLC, LLC, etc.), though to my knowledge none of these sets are available as structured data.
* The W3C Organization Ontology has a [classification](https://www.w3.org/TR/vocab-org/#org:classification) property, but it mentions no specific schemes.
* Schema.org (reusing GoodRelations) has an [eligibleCustomerType](http://schema.org/eligibleCustomerType) property and a list of [common values](http://schema.org/BusinessEntityType).

We could model these classifications as an array of strings, but it seems likely that more robust classification schemes will emerge as structured data, for which an array of Classification objects is better suited.

## Reference

For reference, the full list of Procurement Award Specific Characteristics in DAIMS is:

<details>
<summary>Click to expand</summary>

* 1862 Land Grant College
* 1890 Land Grant College
* 1994 Land Grant College
* 8a Program Participant
* A-76 FAIR Act Action
* Airport Authority
* Alaskan Native Owned Corporation or Firm
* Alaskan Native Servicing Institution
* American Indian Owned Business
* Asian Pacific American Owned Business
* Black American Owned Business
* City Local Government
* Clinger-Cohen Act Planning Compliance
* Commercial Item Acquisition Procedures
* Community Developed Corporation Owned Firm
* Community Development Corporation
* Council of Governments
* Country of Product or Service Origin
* County Local Government
* Commercial Item Test Program
* Consolidated Contract
* Contingency Humanitarian or Peacekeeping Operation
* Contract Bundling
* Contract Financing
* Contracting Officer's Determination of Business Size
* Contracts
* Corporate Entity Not Tax Exempt
* Corporate Entity Tax Exempt
* Cost Accounting Standards Clause
* Cost or Pricing Data
* Davis Bacon Act
* DoD Claimant Program Code
* Domestic Shelter
* DoT Certified Disadvantaged Business Enterprise
* Economically Disadvantaged Women Owned Small Business
* Educational Institution
* Emerging Small Business
* EPA-Designated Products
* Evaluated Preference
* Extent Completed
* Fair Opportunity Limited Sources
* Fed Biz Opps
* Federal Agency
* Federal Action Obligation
* Federally Funded Research and Development Corp
* For Profit Organization
* Foreign Funding
* Foreign Government
* Foreign Owned and Located
* Foundation
* Government Furnished Equipment GFE and Government Furnished Property GFP Grants
* Hispanic American Owned Business
* Hispanic Servicing Institution
* Historically Black College or University
* Historically Underutilized Business Zone HUBZone Firm
* Hospital Flag
* Housing Authorities Public/Trial
* Indian Tribe Federally Recognized
* Information Technology Commercial Item Category
* Interagency Contracting Authority
* Interagency Contracting Category
* Inter-Municipal Local Government
* International Organization
* Interstate Entity
* Joint Venture Economically Disadvantaged Women Owned Small Business
* Joint Venture Women Owned Small Business
* Labor Surplus Area Firm
* Limited Liability Corporation
* Local Area Set Aside
* Local Government Owned
* Major Program
* Manufacturer of Goods
* Minority Institution
* Minority Owned Business
* Multi Year Contract
* Municipality Local Government
* National Interest Action
* Native American Owned Business
* Native Hawaiian Owned Business
* Native Hawaiian Servicing Institution
* Nonprofit Organization
* Number of Actions
* Number of Offers Received
* Other Minority Owned Business
* Other Not For Profit Organization
* Other Statutory Authority
* Other than Full and Open Competition
* Partnership or Limited Liability Partnership
* Performance-Based Service Acquisition
* Place of Manufacture
* Planning Commission
* Port Authority
* Price Evaluation Adjustment Preference Percent Difference
* Private University or College
* Product or Service Code
* Program Acronym
* DOD Acquisition Program
* Purchase Card as Payment Method
* Receives Contracts and Grants
* Recovered Materials/Sustainability
* Research
* SAM Exception
* SBA Certified 8 a Joint Venture
* School District Local Government
* School of Forestry
* Sea Transportation
* Self-Certified Small Disadvantaged Business
* Service Contract Act
* Service Disabled Veteran Owned Business
* Small Agricultural Cooperative
* Small Business Competitiveness Demonstration Program
* Small Disadvantaged Business
* Sole Proprietorship
* Solicitation Identifier
* Solicitation Procedures
* State Controlled Institution of Higher Learning
* Subchapter S Corporation
* Subcontinent Asian Asian - Indian American Owned Business
* Subcontracting Plan
* The AbilityOne Program
* Township Local Government
* Transaction Number
* Transit Authority
* Tribal College
* Tribally Owned Business
* Type Set Aside
* U.S. Federal Government
* U.S. Government Entity
* U.S. State Government
* U.S. Tribal Government
* Veteran Owned Business
* Veterinary College
* Veterinary Hospital
* Walsh Healey Act
* Woman Owned Business
* Women Owned Small Business
* U.S. Local Government
* Domestic or Foreign Entity
* Undefinitized Action

</details>

### PR DESCRIPTION
**Pull request by pindec**
Related #711

### CODE REVIEW

File: docs/guidance/model/organization_classifications.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
+# Organization Classifications
+
+Many publishers wish to disclose and monitor information about different classifications of organizations involved in contracting processes. Example classifications include organization ownership, incorporation, sector, location, and number of employees. We recognise that publishing information about a variety of organization characteristics, for example women-owned and/or indigenous-owned organizations, is an important part of monitoring participation in public procurement.
+
+Some organization classifications, such as organization scale, can be published using a specific field in OCDS. Many other classifications are context-specific, and for this case, we provide an organization classification extension that allows publishers to provide their own classifications.
```
Comment (romifz) [inline docs/guidance/model/organization_classifications.md]: using an specific extension in OCDS?

File: docs/guidance/model/organization_classifications.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+As fields become standardized through the use of option 2, the information can be migrated to _also_ be published via specific fields as in option 1. Publishsers may continue to publish the information in the organization classification extension to preserve backwards compatibility in data sets.
```
Comment (romifz) [inline docs/guidance/model/organization_classifications.md]: via specific extensions (or field extensions?)

Comment (romifz) [reply docs/guidance/model/organization_classifications.md]: Also "Publishers" has a typo here

File: docs/examples/organization-classification/moldova_organization_scale.json [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        "hasEnquiries": false,
+        "awardCriteria": "priceOnly",
+        "submissionMethod": [
+          "electronicSubmission"
+        ],
+        "submissionMethodDetails": "Lista platformelor: achizitii, ebs, licitatie, yptender",
+        "tenderers": [
+          {
+            "id": "MD-IDNO-1012600024777",
+            "name": "Exemplu de companie mică"
```
Comment (romifz) [inline docs/examples/organization-classification/moldova_organization_scale.json]: The name of the party is "Companie mică" above (this is the `legalName`).

File: docs/examples/organization-classification/moldova_organization_scale.json [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+          }
+        ]
+      },
+      "awards": [
+        {
+          "id": "e5b771e0-715e-11e9-94b9-f5c641d7a5b5",
+          "description": "Grupul de lucru anulează procedura de achiziție din considerente că nu există cel puțin trei ofertanți, care au întrunit condițiile de calificare stabilite.",
+          "status": "unsuccessful",
```
Comment (romifz) [inline docs/examples/organization-classification/moldova_organization_scale.json]: I guess this is based on a real example but does it make sense to have a cancelled tender **and** an unsuccessful award?

File: docs/guidance/model/organization_classifications.md [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+
+A third not-recommended example approach using local extensions is also given below, for situations where neither of the above two approaches apply to a specific use case.
+
+## Worked examples
+
+### Option 1: standardized options
+
+#### Organization scale
+
+In the example below, Moldova has disclosed information about the 'Solcoci Prim' organization using the [party scale extension](https://extensions.open-contracting.org/en/extensions/partyScale/master/). The scale is given as 'micro', from the [partyScale.csv codelist](https://extensions.open-contracting.org/en/extensions/partyScale/master/codelists/).
```
Comment (romifz) [inline docs/guidance/model/organization_classifications.md]: The name of the supplier here is different from the example.

File: docs/guidance/model/organization_classifications.md [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+.. jsoninclude:: ../../examples/organization-classification/moldova_organization_scale.json
+   :jsonpointer: /releases/0/parties
+   :expand: details
+   :title: party_scale
+
+```
+
+### Option 2: Organization classification extension
+
+In the examples below, two different publishers have disclosed information about organizations involved in their contracting processes. An organization classification must consist of at least two parts: an identifier for the list (scheme) from which the classification is taken, and an identifier for the category from that list being applied. It is useful to also publish a text label and/or URI that users can draw on to interpret the classification. In the first example, the publisher re-uses an existing classification.scheme. In the second example, where a publisher wishes to track specific policy-related data, a local list of categories is used in preference to mapping to a generic set.
```
Comment (romifz) [inline docs/guidance/model/organization_classifications.md]: We can surround `classification.scheme` with backticks here.

File: docs/guidance/model/organization_classifications.md [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+
+#### Classification schemes
+Each `classification` block contains fields to provide information about the `description` (a textual description or title for the classification code), `id` (the classification code), `uri` (to uniquely identify the classification code) and `scheme`. The `scheme` value can be drawn from the open [`itemClassificationScheme` codelist](../../../schema/codelists/#item-classification-scheme) (where 'Context' is 'organization'), or it can be a local scheme. Schemes are given to classify the activites of procuring authorities (i.e. procuring entities and/or buyers).
```
Comment (romifz) [inline docs/guidance/model/organization_classifications.md]: Typo in "activities"?

File: docs/guidance/model/organization_classifications.md [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+
+Note that for OCDS 1.2, we are considering renaming the 'itemClassificationScheme' codelist to the less-specific 'classificationScheme'.
+
+Where an appropriate scheme is not listed in the [itemClassificationScheme codelist](../../../schema/codelists/#item-classification-scheme), publishers may specify their own scheme. Publishers can either re-use an alternative scheme, or provide their own. Where publishers provide their own local schemes, they should prefix their `scheme` code with an [ISO-3166-1 alpha-3 country code](https://en.wikipedia.org/wiki/ISO_3166-1) to preserve its global uniqueness. Details of this local scheme, and a list of possible codes, should be given in the [publication policy](../publish/publication_policy).
+
+#### Example 2.1 disclosing data using existing schemes
+In the first fictional example below, the UK has disclosed a code from two different EC schemes, 'TED_CE_ACTIVITY' and 'TED_CA_TYPE' to classify the organization whose name is 'London Borough of Haringey'. Refer to the  [`itemClassificationScheme` codelist](../../../schema/codelists/#item-classification-scheme) for further details of these schemes.
```
Comment (romifz) [inline docs/guidance/model/organization_classifications.md]: Should we clarify what "EC" stands for?

File: docs/guidance/model/organization_classifications.md [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+
+For example, although tracking women-owned organizations is shown example 2.2 above, this data only provides information on entities that have been registered as women-owned. Organizations without the classification may be not women-owned, women-owned but not registered as such, or the information may not be known.
+
+To disambiguate these cases, a publisher may choose to publish a flag field for the relevant organization classification. In the fictional example below, Dhanghadi has created a local extension so they can publish data in the `parties.details` block on an organization that is `femaleChaired`, with the values of the field being either 'yes' or 'no'. The publisher would document the structure of this field and its meaning in the local extension files.
```
Comment (romifz) [inline docs/guidance/model/organization_classifications.md]: The example uses `true` as a value for `femaleChaired`, should we change the values of the field to "either true or false"?

File: docs/guidance/model/organization_classifications.md [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+### Option 1: standardized options
+
+#### Organization scale
+
+In the example below, Moldova has disclosed information about the 'Solcoci Prim' organization using the [party scale extension](https://extensions.open-contracting.org/en/extensions/partyScale/master/). The scale is given as 'micro', from the [partyScale.csv codelist](https://extensions.open-contracting.org/en/extensions/partyScale/master/codelists/).
+
+```eval_rst
+
+.. jsoninclude:: ../../examples/organization-classification/moldova_organization_scale.json
+   :jsonpointer: /releases/0/parties
```
Comment (romifz) [inline docs/guidance/model/organization_classifications.md]: I think is better if we can show full releases and just expand the relevant sections, so the readers would have good examples of releases and packages across all the documentation (this is just my opinion).

**Review by romifz — COMMENTED** [review]
Hi @pindec, this looks great. I added a few comments.

File: docs/examples/organization-classification/moldova_organization_scale.json (hunk 4 above)
Comment (pindec) [reply docs/examples/organization-classification/moldova_organization_scale.json]: Good point, I will alter both tender.status and award.status in line with the [example award data](https://github.com/open-contracting/sample-data/blob/master/fictional-example/1.1/ocds-213czf-000-00001-04-award.json).

File: docs/guidance/model/organization_classifications.md (hunk 10 above)
Comment (pindec) [reply docs/guidance/model/organization_classifications.md]: Yes, this would be consistent with the other examples and clearer, thanks.

**Pull request merged**